cpp/benchmarks/ndsh/q01.cpp

```diff
@@ -67,10 +68,14 @@
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref())
 {
-  auto const one = cudf::numeric_scalar<double>(1);
+  auto const one = discount.type().id() == cudf::type_id::DECIMAL64
+                     ? cudf::make_fixed_point_scalar<numeric::decimal64>(1L, numeric::scale_type{0})
+                     : cudf::make_fixed_width_scalar<double>(1);
   auto const one_minus_discount =
-    cudf::binary_operation(one, discount, cudf::binary_operator::SUB, discount.type(), stream, mr);
-  auto const disc_price_type = cudf::data_type{cudf::type_id::FLOAT64};
+    cudf::binary_operation(*one, discount, cudf::binary_operator::SUB, discount.type(), stream, mr);
+  auto const disc_price_type = discount.type().id() == cudf::type_id::DECIMAL64
+                                 ? cudf::data_type{cudf::type_id::DECIMAL64}
+                                 : cudf::data_type{cudf::type_id::FLOAT64};
```

Comment:
Can this just be
```suggestion
  auto const disc_price_type = cudf::data_type{discount.type().id()};
```
?